charts/replicated-library/templates/lib/_pod.tpl

```diff
@@ -72,7 +64,7 @@ volumes:
       {{- /* Add the prefix to the claimName if the claim is in the persistence dict and is enabled */}}
       {{- if and .persistentVolumeClaim .persistentVolumeClaim.claimName }}
         {{- if and (hasKey $.Values.persistence .persistentVolumeClaim.claimName) (get (get $.Values.persistence .persistentVolumeClaim.claimName) "enabled") }}
-          {{- $_ := set .persistentVolumeClaim "claimName" (printf "%s-%s" (include "replicated-library.names.prefix" $) .persistentVolumeClaim.claimName) }}
+          {{- $_ := set .persistentVolumeClaim "claimName" (printf "%s-%s" (include "replicated-library.names.prefix" $) .persistentVolumeClaim.claimName | trimAll "-") }}
```

Comment:
I think you had mentioned this before @chris-sanders , but it seems we'll need to decide what we want to do about name overrides at the object level. I think here it would always use the prefix even if you set fullNameOverride on the object via - https://github.com/replicatedhq/helm-charts/blob/main/charts/replicated-library/templates/lib/_names.tpl#L47. 

Also curious if it should be more clear that we're doing this for the user? If they are using the global fullNameOverride and they expect to set the claimName themselves, i think they'd end up with a duplicate here.

Reply:
I _think_ the way this works, and I tested it when I added it but not since. This is only going to add the prefix if the claim name matches an enabled volume in the persitance api.

If the volume name isn't in the persistance api we don't modify the provided claim name. It's also not modified if the volume is there but disabled.

Assuming that's still working as intended I think it's fine to add the prefix. I guess there is a small chance you have an existing claim of the  same name as a volume you're creating with the chart minus the prefix and you're trying to specify that existing claim. That seems unlikely, but since it is possible I would be fine with a setting to disable this convinence function.

The benefit to this is you don't need to use Advanced Templating just to use a claim on your statefulset which you created via the persistance api.

Reply:
Ok, I think i see what you're saying. Should be fine since if a user decided to do a `fullNameOverride` locally on the PVC then when they change the `claimName` in the `volume` section of the Pod the automation would no longer find an persistence object with a name in the chart matching that override name.